### ISSUE TITLE
False positive for python stdlib `shelve.open()`

### ISSUE THREAD

**Issue by ReagentX** [python/mypy#17209]
**Bug Report**

There is a false positive for python stdlib `shelve.open()` when using a path-like object for the first parameter.

**To Reproduce**

```python
import shelve
from pathlib import Path

shelve.open(Path("filename"))
```

**Expected Behavior**

This should pass mypy, per the Python docs `shelve.open()` accepts a path-like object: https://docs.python.org/3/library/shelve.html#shelve.open

**Actual Behavior**

`my.py:4: error: Argument 1 to "open" has incompatible type "Path"; expected "str"  [arg-type]`

**Your Environment**

- Mypy version used: `mypy 1.10.0 (compiled: yes)`
- Mypy command-line flags: None
- Mypy configuration options from `mypy.ini` (and other config files): Default
- Python version used: 3.12.2
